### PR TITLE
cleans up the remaining new/current_resource ivars in remaining provider code

### PR DESCRIPTION
**Pull request by lamont-granquist**
switches from using ivars to accessors for getters

(and yeah if you're wondering i've thought about adding a chefstyle/rubocop rule to enforce this style...  totally thought about it... ;) )

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@chef/client-core mechanical code change.

**Comment by stevendanna** [comment]
gif-keyboard-12021361866156310935

File: lib/chef/provider/cookbook_file.rb [hunk 1]
```diff
@@ -36,15 +36,15 @@ def initialize(new_resource, run_context)
       end
 
       def load_current_resource
-        @current_resource = Chef::Resource::CookbookFile.new(@new_resource.name)
+        @current_resource = Chef::Resource::CookbookFile.new(new_resource.name)
```
Comment (thommay) [inline lib/chef/provider/cookbook_file.rb]: It saddens me a little that we don't have some sugar around this pattern, given how often we repeat it.

Comment (lamont-granquist) [inline lib/chef/provider/cookbook_file.rb]: we've got `load_current_value` which is that plus more.

**Pull request merged**